### PR TITLE
Organizing code intel docs into explanations, tutorials, how-tos, and references

### PR DESCRIPTION
**Pull request by christinaforney**
Reference: https://docs.google.com/document/d/1bOGK1s6KeIp8CVbYvb3xOsv6k8_XxQQRLOZVcAGbORc/edit#

Related: #14612

There are obviously more improvements that can be made to each of the pages that would help align with the new categorization and organization, but that will have to come next.

### CODE REVIEW

File: doc/user/code_intelligence/tutorials/index.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Tutorials
+
+Tutorials coming soon!
+
+- TODO: Index a popular public repository for Go
+- TODO: Index a popular public repository for TypeScript
```
Comment (efritz) [inline doc/user/code_intelligence/tutorials/index.md:6]: Is this waiting for content pre-merge?

Comment (christinaforney) [reply doc/user/code_intelligence/tutorials/index.md:6]: No, we don't have to wait. What this organization did was help surface the gaps in documentation that we have. These are just my ideas for something we could do. We can pull out the TODOs, or clean this up in a better way, I just wanted to communicate the gap in the PR when I shared with you all.

**Review by efritz — APPROVED** [review]

File: doc/user/code_intelligence/index.md [hunk 2]
```diff
@@ -1,33 +1,62 @@
...
-
-## Basic vs Precise
-
-Basic code intelligence is available by default on all Sourcegraph instances and provides fuzzy code intelligence using a combination of ctags and search. It is great for immediate access to code intelligence features, but due to its dependence on text-based search its results are considered imprecise. 
-
-Precise code intelligence returns metadata from a knowledge graph that is generated through code analysis. The precomputation step is used to generate the graph results in lookups that are fast and have a high degree of accuracy. 
-
-To learn more about how to work with each type of code intelligence visit the [basic](./basic_code_intelligence.md) and [precise](./precise_code_intelligence.md) sections.
+<div class="cta-group">
+<a class="btn btn-primary" href="lsif_quickstart">★ LSIF Quickstart</a>
```
Comment (mrnugget) [inline doc/user/code_intelligence/index.md:10]: If a user doesn't have a clue about code-intel yet and starts by reading this page, at this point they don't know what LSIF is, so I'd remove one barrier-to-click here:

```suggestion
<a class="btn btn-primary" href="lsif_quickstart">★ Quickstart</a>
```

Comment (christinaforney) [reply doc/user/code_intelligence/index.md]: Good point. I don't think it makes sense to have LSIF be the main CTA here. Made intro to code intel the primary and dropped the quickstart down to the how-tos section instead. We can work on creating some new docs that better solve this!

**Comment by christinaforney** [comment]
@efritz - made some changes that touch code (docs references). Can you help check that I didn't break anything?

File: doc/user/code_intelligence/how-to/adding_lsif_to_workflows.md [hunk 3]
```diff
@@ -1,10 +1,10 @@
 # Adding Precise Code Intelligence to CI/CD workflows
 
-## [Language-specific guides](./languages/index.md)
+## [Language-specific guides](././index.md)
```
Comment (efritz) [inline doc/user/code_intelligence/how-to/adding_lsif_to_workflows.md:3]: typo?

**Review by efritz — APPROVED** [review]

File: doc/user/code_intelligence/how-to/adding_lsif_to_workflows.md (hunk 3 above)
Comment (christinaforney) [reply doc/user/code_intelligence/how-to/adding_lsif_to_workflows.md:3]:
```suggestion
## [Language-specific guides](index.md)
```

File: doc/user/code_intelligence/how-to/index.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# How-to guides
+
+- [Add LSIF to many repositories](adding_lsif_to_many_repos.md)
+- [Adding LSIF to CI workflows](adding_lsif_to_workflows.md)
+- [Index a Go repository](index_a_go_repository.md)
+- [Index a TypeScript or JavaScript repository](index_a_typescript_and_javascript_repository.md)
```
Comment (christinaforney) [inline doc/user/code_intelligence/how-to/index.md:6]:
```suggestion
# How-to guides

## Automate uploading LSIF data

- [Add LSIF to many repositories](adding_lsif_to_many_repos.md)
- [Adding LSIF to CI workflows](adding_lsif_to_workflows.md)

## Language-specific guides

- [Index a Go repository](index_a_go_repository.md)
- [Index a TypeScript or JavaScript repository](index_a_typescript_and_javascript_repository.md)
```

File: doc/user/code_intelligence/lsif_quickstart.md [hunk 5]
```diff
@@ -1,6 +1,6 @@
 # LSIF quickstart guide
 
-> NOTE: We are working on creating guides by language, so first check out [language documentation](languages/index.md)! This general guide can be used when a language specific guide is not available.
+> NOTE: We are working on creating guides by language, so first check out [language how-to documentation](how-to/index.md)! This general guide can be used when a language specific guide is not available.
```
Comment (christinaforney) [inline doc/user/code_intelligence/lsif_quickstart.md:3]:
```suggestion
> NOTE: We are working on creating guides by language, so first check out [language documentation](how-to/index.md#language-specific-guides)! This general guide can be used when a language specific guide is not available.
```

File: doc/user/code_intelligence/how-to/adding_lsif_to_workflows.md [hunk 6]
```diff
@@ -177,7 +177,7 @@ The following projects have example Travis CI configurations to generate and upl
 
 ## CI from scratch
 
-If you're indexing a language we haven't documented yet in our [language-specific guides](languages/index.md), follow the instructions in this section. We want to have containers available for every language with a robust indexer, so please also consider [filing an issue](https://github.com/sourcegraph/sourcegraph/issues/new) to let us know we're missing one.
+If you're indexing a language we haven't documented yet in our [language-specific guides](./index.md), follow the instructions in this section. We want to have containers available for every language with a robust indexer, so please also consider [filing an issue](https://github.com/sourcegraph/sourcegraph/issues/new) to let us know we're missing one.
```
Comment (christinaforney) [inline doc/user/code_intelligence/how-to/adding_lsif_to_workflows.md:180]:
```suggestion
If you're indexing a language we haven't documented yet in our [language-specific guides](./index.md#language-specific-guides), follow the instructions in this section. We want to have containers available for every language with a robust indexer, so please also consider [filing an issue](https://github.com/sourcegraph/sourcegraph/issues/new) to let us know we're missing one.
```

**Pull request merged**